### ISSUE TITLE
Show mount points

### ISSUE THREAD

**Issue by Ape** [sxyazi/yazi#1272]
### Please describe the problem you're trying to solve

I'd like to see if a directory is a mount point.

### Would you be willing to contribute this feature?

- [ ] Yes, I'll give it a shot

### Describe the solution you'd like

It could look something like this:

```
bin
boot (vfat on /dev/sdb1)
dev (devtmpfs)
etc
home (btrfs on /dev/sdc1)
lib
lib64
mnt
opt
proc (proc)
root
run (tmpfs)
sbin
srv
sys (sysfs)
tmp (tmpfs)
usr
var
```

**Comment by sxyazi** [comment]
Does this mean it only works on Unix systems? I'm not sure if Windows has the concept of mount points.

I think it should be implemented as a plugin rather than a builtin feature. Recently, I've been enhancing the API for extending UI components with plugins, https://github.com/sxyazi/yazi/pull/1257. This will make it easier to implement it as a UI plugin & fetcher.

**Comment by Ape** [comment]
Yeah, I guess this would be Unix only. I'd be happy to have the feature as a plugin.

**Comment by Ape** [comment]
I made a proof-of-concept plugin for this feature: https://github.com/Ape/mount-points.yazi

However, it is currently quite unusable because of performance implications. The plugin refreshes the mount point information constantly. Ideally it would do it in a task and only when needed.

**Comment by AnirudhG07** [comment]
@Ape there's a plugin [archivemount.yazi](https://github.com/AnirudhG07/archivemount.yazi), it can mount archives(as the name suggests) as well tell if a direc is a mountpoint or not.
Check if out. If you do not find it satisfactory, we can update the UI part for that plugin, as per requirements suggest. Thanks!

**Comment by Ape** [comment]
> @Ape there's a plugin [archivemount.yazi](https://github.com/AnirudhG07/archivemount.yazi), it can mount archives(as the name suggests) as well tell if a direc is a mountpoint or not. Check if out. If you do not find it satisfactory, we can update the UI part for that plugin, as per requirements suggest. Thanks!

That plugin doesn't seem to currently have the feature I'm looking for. Check `mount-points.yazi` for reference.

**Comment by ovwxxwvo** [comment]
@Ape @sxyazi hey, mate ! how about this. I thing this will be a little bit better.
![20241118_120238](https://github.com/user-attachments/assets/af80aacf-2542-46f2-a0bb-feb545c5b339)
```
function Status:name()
	local h = self._tab.current.hovered
	if not h then return ui.Line {} end

  local mounted = io.popen("findmnt -n --output SOURCE,FSTYPE " .. tostring(h.url)):read("*a")
 	if mounted ~= "" then
 		mounted = " <- " .. mounted
 	  return ui.Line(" " .. h.name .. mounted)
    end

 	local linked = ""
 	if h.link_to ~= nil then
 		linked = " -> " .. tostring(h.link_to)
 	  return ui.Line(" " .. h.name .. linked)
 	  end

  end
```
